gost/microsoft.go

```diff
@@ -175,169 +176,214 @@ func (ms Microsoft) DetectCVEs(r *models.ScanResult, _ bool) (nCVEs int, err err
 
-						vera, err := version.NewVersion(p.Version)
-						if err != nil {
-							kbs = append(kbs, kb)
-							break
+							vera, err := version.NewVersion(p.Version)
+							if err != nil {
+								kbs = append(kbs, kb)
+								break
+							}
+							verb, err := version.NewVersion(kb.FixedBuild)
+							if err != nil {
+								kbs = append(kbs, kb)
+								break
+							}
+							if vera.LessThan(verb) {
+								kbs = append(kbs, kb)
+							}
+						default:
+							// kbs = append(kbs, kb)
 						}
-						verb, err := version.NewVersion(kb.FixedBuild)
-						if err != nil {
-							kbs = append(kbs, kb)
-							break
+					} else {
+						if slices.Contains(applied, kb.Article) {
+							return nil
 						}
-						if vera.LessThan(verb) {
+						if slices.Contains(unapplied, kb.Article) {
 							kbs = append(kbs, kb)
 						}
 					}
-				} else {
-					if slices.Contains(applied, kb.Article) {
-						kbs = []gostmodels.MicrosoftKB{}
-						break
-					}
-					if slices.Contains(unapplied, kb.Article) {
-						kbs = append(kbs, kb)
-					}
 				}
-			}
-			if len(kbs) > 0 {
-				p.KBs = kbs
+				return kbs
+			}()
+			if len(p.KBs) > 0 {
 				ps = append(ps, p)
 			}
 		}
-		cve.Products = ps
-		if len(cve.Products) == 0 {
-			continue
-		}
-		nCVEs++
+		return ps
+	}()
+	if len(cve.Products) == 0 {
+		return nil, nil
+	}
 
-		cveCont, mitigations := ms.ConvertToModel(&cve)
-		uniqKB := map[string]struct{}{}
-		var stats models.PackageFixStatuses
-		for _, p := range cve.Products {
-			for _, kb := range p.KBs {
-				if _, err := strconv.Atoi(kb.Article); err != nil {
-					switch {
-					case strings.HasPrefix(p.Name, "Microsoft Edge"):
-						s := models.PackageFixStatus{
-							Name:     "Microsoft Edge",
-							FixState: "fixed",
-							FixedIn:  kb.FixedBuild,
-						}
-						if kb.FixedBuild == "" {
-							s.FixState = "unknown"
-						}
-						stats = append(stats, s)
-					default:
-						stats = append(stats, models.PackageFixStatus{
-							Name:     p.Name,
-							FixState: "unknown",
-							FixedIn:  kb.FixedBuild,
-						})
-					}
-				} else {
-					uniqKB[fmt.Sprintf("KB%s", kb.Article)] = struct{}{}
-				}
+	cveCont, mitigations := ms.ConvertToModel(&cve)
+	vinfo := models.VulnInfo{
+		CveID:       cve.CveID,
+		CveContents: models.NewCveContents(*cveCont),
+		Mitigations: mitigations,
+	}
+
+	for _, p := range cve.Products {
+		if len(p.KBs) == 0 {
+			switch {
+			case p.Name == r.Release:
+				vinfo.AffectedPackages = append(vinfo.AffectedPackages, models.PackageFixStatus{
+					Name:     p.Name,
+					FixState: "unfixed",
+				})
+			case strings.HasPrefix(p.Name, "Microsoft Edge"):
+				vinfo.AffectedPackages = append(vinfo.AffectedPackages, models.PackageFixStatus{
+					Name:     "Microsoft Edge",
+					FixState: "unknown",
+				})
+			default:
+				// vinfo.AffectedPackages = append(vinfo.AffectedPackages, models.PackageFixStatus{
+				// 	Name:     p.Name,
+				// 	FixState: "unknown",
+				// })
```

Comment:
Isn't it unnecessary?

Reply:
Here, because the KBID is unknown, it explicitly states that products other than Microsoft Edge will be ignored when determining whether they are affected by the FixedBuild.

https://github.com/future-architect/vuls/pull/2041/files#diff-ac7ec2f2b5b040d5c39ebcb508a3c338fa3ca63c191858397ecb042b7b082c43R231

Even in the case of no KB, i.e., suspected unfixed, it is commented out to clearly indicate that it does not support anything other than Windows OS and Microsoft Edge.